compiler/codegen/type.cpp

```diff
@@ -423,10 +430,25 @@ void AggregateType::codegenDef() {
                                           symbol->cname);
         }
 
-
         llvm::StructType* stype = llvm::cast<llvm::StructType>(type);
         stype->setBody(params);
 
+        if (aligns.size() == params.size()) {
+          for (int i = 0; i < params.size(); i++) {
+            unsigned offset = info->module->getDataLayout().getStructLayout(stype)->getElementOffset(i);
+#if HAVE_LLVM_VER >= 100
+            unsigned align = 1 << Log2(aligns[i].valueOrOne());
```

Comment:
What's the point of `1 << Log2(x)` ? Isn't it the same as `unsigned align = aligns[i].valueOrOne()` ?

Reply:
`MaybeAlign.valueOrOne()` returns an `Align` class. `Align` has a `value` method, but its documentation says "This is a hole in the type system and should not be abused." so I used `Log2` instead to get the alignment out as an unsigned int.